src/main/kotlin/io/github/gradlenexus/publishplugin/GetStagingProfile.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import org.gradle.api.GradleException
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import javax.inject.Inject
+
+/**
+ * Gradle task to retrieve the staging profile id for a given [repository] and [packageGroup]
+ */
+@Suppress("UnstableApiUsage")
+open class GetStagingProfile @Inject constructor(
+    objects: ObjectFactory,
+    extension: NexusPublishExtension,
+    repository: NexusRepository
+) : AbstractNexusStagingRepositoryTask(objects, extension, repository) {
+
+    /**
+     * This property is [Optional] because it is only required if [NexusRepository.stagingProfileId] is not provided
+     */
+    @Input
+    @Optional
+    val packageGroup = objects.property<String>().apply {
+        set(extension.packageGroup)
+    }
+
+    @OutputFile
+    val stagingProfileIdOutput: RegularFileProperty = objects.fileProperty()
```

Comment:
I'm unfamiliar with the `@Output` annotation - could you point me to some documentation? I don't see it in the `org.gradle.api.tasks` package and was under the impression that task outputs generally were associated with a file/directory

Reply:
1. `@Output` does not exist
2. By default, `Gradle` would consider the task `UP-TO-DATE` if all inputs and outputs are the same.

It looks like `OutputFile` is OK here.

Reply:
I don't think this task should ever be up-to-date since it depends on state on the server. Thus, if we remove the `@OutputFile` property and add an `@Internal val stagingProfileId: Property<String>` we don't need to write a file.

Reply:
Sorry @ryandens for twisting the things around. I liked the idea with dependent tasks, however, the recent comments by Marc helped me realize that fetching the staging repository id from a server for diagnostic purpose is not the same as resolving staging repository id for the initialize tasks (and others). In the first case, server should be always asked (should ignore configured stagingProfileId), output should be printed on lifecycle level (people always want to see it) and it should be always not up-to-date. On the other side, for the initialize task is perfectly fine to get pre-configured value (if available), be quiet and play nicely with caching (assuming that `stagingProfileId` was set).

As a result, I think that we need `Retrieve<<Repo>>StagingProfile` task to implement the first case (always using  `client.findStagingProfileId()`). The second one could be covered as in the original version in `InitializeNexusStagingRepository`. As it is recommended to set `stagingProfileId` in the project configuration (for the performance reasons), I don't really see practical use case where someone would need to get that value from server on its own to reuse (with `stagingRepositoryId` the situation is completely different). WDYT?

Reply:
@ryandens Are you ok with the proposed modifications (simplifications)?

Reply:
Yes, this seems reasonable! I'll take a stab at getting this all sorted out